src/app/pages/checkout-payment/payment-concardis-creditcard-cvc-detail/payment-concardis-creditcard-cvc-detail.component.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { takeUntil } from 'rxjs/operators';
+
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+import { PaymentInstrument } from 'ish-core/models/payment-instrument/payment-instrument.model';
+import { ScriptLoaderService } from 'ish-core/utils/script-loader/script-loader.service';
+import { markAsDirtyRecursive } from 'ish-shared/forms/utils/form-utils';
+
+import { PaymentConcardisComponent } from '../payment-concardis/payment-concardis.component';
+
+// allows access to concardis js functionality
+// tslint:disable-next-line:no-any
+declare var PayEngine: any;
+
+@Component({
+  selector: 'ish-payment-concardis-creditcard-cvc-detail',
+  templateUrl: './payment-concardis-creditcard-cvc-detail.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+// tslint:disable-next-line: rxjs-prefer-angular-takeuntil
+export class PaymentConcardisCreditcardCvcDetailComponent extends PaymentConcardisComponent implements OnInit {
+  @Input() paymentInstrument: PaymentInstrument;
+  @Output() disableNextStep = new EventEmitter<boolean>();
+  validityTimeInMinutes: string;
+  cvcDetailForm: FormGroup;
+
+  constructor(
+    protected scriptLoader: ScriptLoaderService,
+    protected cd: ChangeDetectorRef,
+    private checkoutFacade: CheckoutFacade
+  ) {
+    super(scriptLoader, cd);
+    this.cvcDetailForm = new FormGroup({
+      cvcDetail: new FormControl(undefined, [Validators.required]),
+    });
+  }
+
+  /**
+   * load concardis script if component is visible
+   */
+  loadScript() {
+    // load script only once if component becomes visible
+    if (!this.scriptLoaded) {
+      const merchantId = this.getParamValue(
+        'ConcardisPaymentService.MerchantID',
+        'checkout.credit_card.merchantId.error.notFound'
+      );
+      this.validityTimeInMinutes = this.getParamValue(
```

Comment:
Is this code right here? Maybe it has to be executed also, if the script has already been loaded by another concardis component